Content.Server/Shipyard/Systems/ShipyardSystem.Consoles.cs

```diff
@@ -0,0 +1,176 @@
+        [Dependency] private readonly UserInterfaceSystem _ui = default!;
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly ShipyardSystem _shipyard = default!;
+        [Dependency] private readonly StationSystem _station = default!;
+        [Dependency] private readonly CargoSystem _cargo = default!;
+        [Dependency] private readonly RadioSystem _radio = default!;
+        [Dependency] private readonly SharedAudioSystem _audio = default!;
+
+        public void InitializeConsole()
+        {
+            SubscribeLocalEvent<ShipyardConsoleComponent, ShipyardConsolePurchaseMessage>(OnPurchaseMessage);
+            SubscribeLocalEvent<ShipyardConsoleComponent, BoundUIOpenedEvent>(OnConsoleUIOpened);
+            //SubscribeLocalEvent<ShipyardConsoleComponent, ComponentInit>(OnInit);
+            SubscribeLocalEvent<RoundRestartCleanupEvent>(Reset);
+        }
+
+        private void OnInit(EntityUid uid, SharedShipyardConsoleComponent orderConsole, ComponentInit args)
+        {
+            //_shipyard.SetupShipyard(); ///if we have to start up the shipyard from here later
+        }
+
+        private void Reset(RoundRestartCleanupEvent ev)
+        {
+            //_shipyard.Shutdown(); //round cleanup event in case of needing OnInit;
+        }
+
+        private void OnPurchaseMessage(EntityUid uid, SharedShipyardConsoleComponent component, ShipyardConsolePurchaseMessage args)
+        {
+            if (args.Session.AttachedEntity is not { Valid : true } player)
+            {
+                return;
+            }
+
+            if (TryComp<AccessReaderComponent>(uid, out var accessReaderComponent) && accessReaderComponent.Enabled && !_access.IsAllowed(player, accessReaderComponent))
+            {
+                ConsolePopup(args.Session, Loc.GetString("comms-console-permission-denied"));
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            VesselPrototype? vessel = null;
+
+            if (!_prototypeManager.TryIndex<VesselPrototype>(args.Vessel, out vessel) || vessel == null)
+            {
+                ConsolePopup(args.Session, Loc.GetString("shipyard-console-invalid-vessel", ("vessel", args.Vessel)));
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            if (vessel.Price <= 0)
+                return;
+
+            var station = _station.GetOwningStation(uid);
+            var bank = GetBankAccount(station);
+
+            if (bank == null)
+                return;
+
+            if (bank.Balance <= vessel.Price)
+            {
+                ConsolePopup(args.Session, Loc.GetString("cargo-console-insufficient-funds", ("cost", vessel.Price)));
+                PlayDenySound(uid, component);
+                return;
+            }
+
+            if (!TryPurchaseVessel(bank, vessel, out var shuttle) || shuttle == null)
```

Comment:
TryPurchaseVessel should use the NotNullWhen attribute for it's output so that the null check is redundant.